automarker_2/automarker_app/lib/automarker_test_runner.py

```diff
@@ -182,10 +186,47 @@ def register_test_error(self, error_message, status=STEP_STATUS_NOT_YET_COMPETEN
             }
         )
 
+    def assert_no_repeated_tags(self):
+        assert (
+            len(self.last_command_output.repeating_tags()) == 0
+        ), f"expected no repeating tags but the following tags appear more than once: {self.last_command_output.repeating_tags()}"
+
+    def assert_no_unfinished_tags(self):
+        if not self.last_command_output.stderr:
+            assert (
+                len(self.last_command_output.unfinished_tags()) == 0
```

Comment:
I think the assertion message is a bit confusing. How about something like `expected tags to be closed but the following tags appear to be left open:...`